perfscale_regression_ci/scripts/scalability/pod-affinity-anti-affinity.sh

```diff
@@ -0,0 +1,113 @@
+#/!/bin/bash
+################################################
+## Auth=prubenda@redhat.com lhorsley@redhat.com
+## Desription: This testcase tests Pod Affinity and anti-affinity as we approach pod capacity on compute nodes.
+## Polarion test case: OCP-18083 - NextGen Pod scheduler at capacity with Pod Affinity and Anti-Affinity rules	
+## https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-18083
+## Cluster config: Cluster needed in AWS EC2 (m5.xlarge, 3 master/etcd, 3 worker/compute nodes)
+## kube-burner config: perfscale_regerssion_ci/kubeburner-object-templates/pod-affinity-anti-affinity-config.yml
+################################################ 
+
+
+source ../../utils/run_workload.sh
+source ../custom_workload_env.sh
+source ../common.sh
+source pod-affinity-anti-affinity_env.sh
+
+
+export POD_ANTI_AFFINITY_JOB_ITERATION=${POD_ANTI_AFFINITY_JOB_ITERATION:-190}
+export POD_AFFINITY_JOB_ITERATION=${POD_AFFINITY_JOB_ITERATION:-190}
+
+
+# Output some general information about the test environment
+date
+uname -a
+oc get clusterversion
+oc version
+oc get node --show-labels
+oc describe node | grep Runtime
+
+
+echo "======Setup/Configuration: Create the s1-proj namespace and the s1-test pod======"
+
+worker_nodes=$(oc get nodes -l 'node-role.kubernetes.io/worker=' | awk '{print $1}' | grep -v NAME | xargs)
+
+echo -e "\nWorker  nodes are: $worker_nodes"
+
+oc get nodes -l 'node-role.kubernetes.io/worker='
+oc describe nodes -l 'node-role.kubernetes.io/worker='
+
+
+echo "======Use kube-burner to load the cluster with test objects======"
+run_workload
+
+echo "======Checking the pods for errors======"
+
+wait_for_running $S1_PROJ_NAMESPACE
+check_no_error_pods $S1_PROJ_NAMESPACE
+
+wait_for_running $POD_AFFINTIY_NAMESPACE
+check_no_error_pods $POD_AFFINTIY_NAMESPACE
+
+wait_for_running $POD_ANTI_AFFINTIY_NAMESPACE
+check_no_error_pods $POD_ANTI_AFFINTIY_NAMESPACE
+
+# Once the pod is running, find it's node
+s1pod_node=$(oc get pods -n s1-proj -o wide --no-headers | awk {'print $7}')
+
+echo -e "\nPod s1 was deployed on node ${s1pod_node}"
+
+echo -e "\n============= Summary of pod count with affinity to s1 pod: =================="
+
+s1_affinity_pod_expected=$POD_AFFINITY_JOB_ITERATION
+s1_anti_affinity_pod_expected=$POD_ANTI_AFFINITY_JOB_ITERATION
+
+
+
+s1_affinity_pod_actual=$(oc get pods -n ${POD_AFFINTIY_NAMESPACE} -o wide | grep ${s1pod_node} | grep Running | wc -l | xargs)
+echo -e "\nNumber of pods deployed with pod affinity to pod s1:  ${s1_affinity_pod_actual} , expecting ${s1_affinity_pod_expected} pods"
+
+echo -e "\n============= Summary of pod count with anit-affinity to s1 pod: =================="
+
+
+
+s1_anti_affinity_pod_actual=$(oc get pods -n ${POD_ANTI_AFFINTIY_NAMESPACE} -o wide | grep -v ${s1pod_node} | grep Running | wc -l | xargs)
```

Comment:
Let's create a parameterized function that takes in a namespace and a node name and returns a number of pods because we are doing a very similar thing 4 times amongst 2 scripts 

Also update this line to use this function please
https://github.com/openshift/svt/blob/df19c2406207ce35cfa518a44e1aa5b625a9e284/perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh#L104

Reply:
Done. I'll address the change to the node-affinity-anti-affinity test in a different PR.